### PR TITLE
Adapt test result screens to include family tests (EXPOSUREAPP-12318,12319,12320)

### PR DESCRIPTION
**Pull request by SamuraiKek**
Text changes and conditions for displaying test results for family tests. The cards displayed in the result screens will be adapted in a different PR.

How to test:

When scanning a pcr/rat test, the result screen will display the right information depending on which type of test you selected after scanning the QR (For me/For others). If you choose to scan the test for others, the test result screen will show up with the name of the person the test belongs to and a few text changes. When returning back to the home screen, the tile for family tests will be visible (selecting it will lead to a blank screen for now).

### CODE REVIEW

**Comment by mtwalli** [comment]
I think we need to modify the logic a bit, not just xml changes. Currently view models for test results screens rely on test type, which is not suitable any longer in this case, I think we need to use test identifier and look for the test in both repos

**Review by CV113 — DISMISSED** [review]
UA approved

**Comment by mtwalli** [comment]
@SamuraiKek Did you update the screenshots ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/CoronaTestProvider.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import timber.log.Timber
+import javax.inject.Inject
+
+class CoronaTestProvider @Inject constructor(
+    private val coronaTestRepository: CoronaTestRepository,
+    private val familyTestRepository: FamilyTestRepository,
+    private val testResultAvailableNotificationService: PCRTestResultAvailableNotificationService,
+) {
+
+    private val allTests: Flow<Set<BaseCoronaTest>> = combine(
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/CoronaTestProvider.kt:21]: Add shareLatest

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/CoronaTestProvider.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        coronaTestRepository.coronaTests,
+        familyTestRepository.familyTests
+    ) { personalCoronaTests, familyCoronaTests ->
+        personalCoronaTests.plus(familyCoronaTests)
+    }
+
+    fun findTestById(testIdentifier: TestIdentifier): Flow<BaseCoronaTest?> =
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/CoronaTestProvider.kt:28]: What about renaming this fun like testForIdentifier?

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/submission/testavailable/SubmissionTestResultAvailableViewModelTest.kt [hunk 3]
```diff
@@ -121,7 +121,7 @@ class SubmissionTestResultAvailableViewModelTest : BaseTest() {
     }
 
     @Test
-    fun `go to test result without updating TEK history if NO consent is given`() {
+    suspend fun `go to test result without updating TEK history if NO consent is given`() {
```
Comment (BMItr) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/submission/testavailable/SubmissionTestResultAvailableViewModelTest.kt:124]: RunBlockingTest instead making it a suspend fun

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/CoronaTestProvider.kt (hunk 1 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/CoronaTestProvider.kt:21]: is is private , used internally , no need to share , I actually I added it

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/CoronaTestProvider.kt (hunk 2 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/CoronaTestProvider.kt:28]: what difference would it make ? `findTestById` is self explanatory or? you think it is confusing

**Review by BMItr — DISMISSED** [review]
lgtm! please check tests. (will reapprove after update)

**Review by mtwalli — APPROVED** [review]

**Pull request merged**